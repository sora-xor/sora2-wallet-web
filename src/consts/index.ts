import type { FPNumber, Operation } from '@sora-substrate/util';

export const HiddenValue = '******';

export const BLOCK_PRODUCE_TIME = 6_000;
export const MAX_ALERTS_NUMBER = 5;

export enum AppWallet {
  GoogleDrive = 'google-drive',
  FearlessWallet = 'fearless-wallet',
  PolkadotJS = 'polkadot-js',
  SubwalletJS = 'subwallet-js',
  TalismanJS = 'talisman',
}

export enum AccountActionTypes {
  Rename = 'rename',
  Export = 'export',
  Logout = 'logout',
  Delete = 'delete',
}

export enum RouteNames {
  InternalConnection = 'InternalConnection',
  WalletConnection = 'WalletConnection',
  WalletSend = 'WalletSend',
  Wallet = 'Wallet',
  WalletAssetDetails = 'WalletAssetDetails',
  CreateToken = 'CreateToken',
  ReceiveToken = 'ReceiveToken',
  AddAsset = 'AddAsset',
  SelectAsset = 'SelectAsset',
}

export enum WalletTabs {
  Assets = 'WalletAssets',
  History = 'WalletHistory',
}

export enum TokenTabs {
  Token = 'CreateSimpleToken',
  NonFungibleToken = 'CreateNftToken',
}

export enum AddAssetTabs {
  Token = 'AddAssetToken',
  NFT = 'AddAssetNFT',
}

export enum WalletFilteringOptions {
  All = 'All',
  Currencies = 'Currencies',
  NFT = 'NFT',
}

export enum SoraNetwork {
  Dev = 'Dev',
  Test = 'Test',
  Stage = 'Stage',
  Prod = 'Prod',
}

export enum HashType {
  ID = 'id',
  Block = 'block',
  Account = 'account',
  EthAccount = 'ethAccount',
  EthTransaction = 'ethTransaction',
}

export enum ExplorerType {
  Sorascan = 'sorascan',
  Subscan = 'subscan',
  Polkadot = 'polkadot',
}

export type ExplorerLink = {
  type: ExplorerType;
  value: string;
};

export enum Step {
  CreateSimpleToken = 'CreateSimpleToken',
  ConfirmSimpleToken = 'ConfirmSimpleToken',
  CreateNftToken = 'CreateNftToken',
  ConfirmNftToken = 'ConfirmNftToken',
  Warn = 'Warn',
}

export enum LoginStep {
  Welcome = 'Welcome',
  Import = 'Import',
  ImportCredentials = 'Import/Credentials',
  ImportExternalExtensionList = 'ImportExternal/ExtensionList',
  ImportExternalAcccountList = 'ImportExternal/AccountList',
  SeedPhrase = 'Create/SeedPhrase',
  ConfirmSeedPhrase = 'Create/ConfirmSeedPhrase',
  CreateCredentials = 'Create/Credentials',
  AccountList = 'AccountList',
}

export const AccountImportInternalFlow = [LoginStep.Import, LoginStep.ImportCredentials];
export const AccountImportExternalFlow = [LoginStep.ImportExternalExtensionList, LoginStep.ImportExternalAcccountList];
export const AccountCreateFlow = [LoginStep.SeedPhrase, LoginStep.ConfirmSeedPhrase, LoginStep.CreateCredentials];

export interface WalletPermissions {
  addAssets?: boolean;
  addLiquidity?: boolean;
  bridgeAssets?: boolean;
  createAssets?: boolean;
  sendAssets?: boolean;
  showAssetDetails?: boolean;
  swapAssets?: boolean;
}

export type EthBridgeUpdateHistory = (updateWalletHistory: VoidFunction) => Promise<void>;

export interface WalletAssetFilters {
  option: WalletFilteringOptions;
  verifiedOnly: boolean;
  zeroBalance: boolean;
}

export interface WalletInitOptions {
  withoutStore?: boolean;
  permissions?: WalletPermissions;
  updateEthBridgeHistory?: EthBridgeUpdateHistory;
}

export interface NetworkFeeWarningOptions {
  type: Operation;
  isXor?: boolean;
  amount?: FPNumber;
}

export enum FontSizeRate {
  SMALL = 'small',
  MEDIUM = 'medium',
  NORMAL = 'normal',
}

export enum FontWeightRate {
  SMALL = 'small',
  MEDIUM = 'medium',
  NORMAL = 'normal',
}

export enum LogoSize {
  MINI = 'mini',
  SMALL = 'small',
  MEDIUM = 'medium',
  BIG = 'big',
  BIGGER = 'bigger',
  LARGE = 'large',
}

export enum PaginationButton {
  First = 'first',
  Prev = 'prev',
  Next = 'next',
  Last = 'last',
}

// states for Ethereum Bridge transaction
export enum ETH_BRIDGE_STATES {
  INITIAL = 'INITIAL',
  SORA_SUBMITTED = 'SORA_SUBMITTED',
  SORA_PENDING = 'SORA_PENDING',
  SORA_REJECTED = 'SORA_REJECTED',
  SORA_COMMITED = 'SORA_COMMITED',
  EVM_SUBMITTED = 'EVM_SUBMITTED',
  EVM_PENDING = 'EVM_PENDING',
  EVM_REJECTED = 'EVM_REJECTED',
  EVM_COMMITED = 'EVM_COMMITED',
}

export const ObjectInit = () => null;

/**
 * DO NOT IMPORT THIS CONST!
 *
 * Contains wallet-specific words which shouldn't be translated.
 * It's used in TranslationMixin of SORA Wallet project and it's extended in Polkaswap TranslationMixin.
 */
export const TranslationConsts = {
  Ethereum: 'Ethereum',
  Etherscan: 'Etherscan',
  Hashi: 'HASHI',
  PolkadotJs: 'Polkadot{.js}',
  Sora: 'SORA',
  TBC: 'TBC',
  XYK: 'XYK',
  NFT: 'NFT',
  Polkadot: 'Polkadot',
  SORAScan: 'SORAScan',
  Subscan: 'Subscan',
  QR: 'QR', // QR code
  IPFS: 'IPFS', // Inter-planetary file system
  // SORA networks
  soraNetwork: {
    Dev: 'SORA Devnet',
    Test: 'SORA Testnet (private)',
    Stage: 'SORA Testnet',
    Prod: 'SORA Mainnet',
  },
  JSON: '.json',
} as const;
