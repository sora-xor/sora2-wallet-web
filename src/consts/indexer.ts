export const wsClientRetryAttempts = 10;
export const wsClientLazy = true;
export const wsClientReconnect = true;
