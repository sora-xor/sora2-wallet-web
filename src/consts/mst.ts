export const mstTrxDeadline = {
  '1D': 14400, // 1 Day
  '7D': 100800, // 7 Days
  '14D': 201600, // 14 Days
  '30D': 432000, // 30 Days
};
