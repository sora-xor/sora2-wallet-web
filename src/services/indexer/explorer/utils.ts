import { formatStringNumber } from '../../../util';

import type { SubqueryAssetEntity } from '../subquery/types';
import type { SubsquidAssetEntity } from '../subsquid/types';
import type { FiatPriceObject, UpdatesStream, PoolApyObject } from '../types';

export function parseAssetFiatPrice(entity: SubsquidAssetEntity | SubqueryAssetEntity): FiatPriceObject {
  const acc = {};
  const id = entity.id;
  const priceFPNumber = formatStringNumber(entity.priceUSD);
  const isPriceFinity = priceFPNumber.isFinity();
  if (isPriceFinity) {
    acc[id] = priceFPNumber.toCodecString();
  }
  return acc;
}

export function parsePriceStreamUpdate(entity: UpdatesStream): Nullable<FiatPriceObject> {
  if (!entity?.data) return null;

  const data = JSON.parse(entity.data);

  return Object.entries(data).reduce((acc, [id, price]) => {
    const priceFPNumber = formatStringNumber(price as string);
    const isPriceFinity = priceFPNumber.isFinity();
    if (isPriceFinity) {
      acc[id] = priceFPNumber.toCodecString();
    }
    return acc;
  }, {});
}

export function parseApyStreamUpdate(entity: UpdatesStream): PoolApyObject {
  const data = entity?.data ? JSON.parse(entity.data) : {};

  return Object.entries(data).reduce((acc, [id, apy]) => {
    const strategicBonusApyFPNumber = formatStringNumber(apy as string);
    const isStrategicBonusApyFinity = strategicBonusApyFPNumber.isFinity();
    if (isStrategicBonusApyFinity) {
      acc[id] = strategicBonusApyFPNumber.toCodecString();
    }
    return acc;
  }, {});
}
