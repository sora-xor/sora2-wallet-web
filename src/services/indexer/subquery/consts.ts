export const SubstrateEvents = {
  CurrenciesTransferred: {
    method: 'Transferred',
    section: 'currencies',
  },
};
