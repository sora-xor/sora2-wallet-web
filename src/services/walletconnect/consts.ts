export const WALLET_CONNECT_PROJECT_ID = 'feeab08b50e0d407f4eb875d69e162e8';
