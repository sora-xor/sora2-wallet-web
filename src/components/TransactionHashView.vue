<template>
  <div class="s-input-container">
    <s-input :placeholder="t(translation)" :value="formattedAddress" readonly tabindex="-1" />
    <s-button
      class="s-button--copy"
      icon="basic-copy-24"
      :tooltip="copyTooltip(t(translation))"
      type="action"
      alternative
      @click="handleCopyAddress(formattedValue, $event)"
    />
    <s-dropdown
      class="s-dropdown-menu"
      borderRadius="mini"
      type="ellipsis"
      icon="basic-more-vertical-24"
      placement="bottom-end"
      @select="isEthHash ? handleOpenEtherscan() : undefined"
    >
      <template slot="menu">
        <s-dropdown-item v-if="isEthHash" class="s-dropdown-menu__item">
          <span>{{ t('bridgeTransaction.viewInEtherscan') }}</span>
        </s-dropdown-item>
        <template v-else>
          <a
            v-for="link in explorerLinks"
            :key="link.type"
            class="transaction-link"
            :href="link.value"
            target="_blank"
            rel="nofollow noopener"
          >
            <s-dropdown-item class="s-dropdown-menu__item">
              {{ t(`transaction.viewIn.${link.type}`) }}
            </s-dropdown-item>
          </a>
        </template>
      </template>
    </s-dropdown>
  </div>
</template>

<script lang="ts">
import { Component, Mixins, Prop } from 'vue-property-decorator';

import { formatAddress, getExplorerLinks, formatSoraAddress } from '../util';
import TranslationMixin from './mixins/TranslationMixin';
import CopyAddressMixin from './mixins/CopyAddressMixin';
import { ExplorerLink, SoraNetwork, HashType, ExplorerType } from '../consts';
import { state } from '../store/decorators';

@Component
export default class TransactionHashView extends Mixins(TranslationMixin, CopyAddressMixin) {
  @Prop({ type: String, required: true }) readonly value!: string;
  @Prop({ type: String, required: true }) readonly type!: HashType;
  @Prop({ type: String, required: true }) readonly translation!: string;
  @Prop({ type: String, default: '' }) readonly hash!: string;

  @state.settings.soraNetwork private soraNetwork!: SoraNetwork;

  get isEthHash(): boolean {
    return [HashType.EthAccount, HashType.EthTransaction].includes(this.type);
  }

  get formattedValue(): string {
    if (this.type === HashType.Account) {
      return formatSoraAddress(this.value);
    }
    return this.value;
  }

  get displayValue(): string {
    return this.hash || this.formattedValue;
  }

  get explorerLinks(): Array<ExplorerLink> {
    if (this.isEthHash) return [];

    const baseLinks = getExplorerLinks(this.soraNetwork);
    if ([HashType.Account, HashType.Block].includes(this.type)) {
      return baseLinks.map(({ type, value }) => ({ type, value: `${value}/${this.type}/${this.formattedValue}` }));
    }
    return baseLinks.map(({ type, value }) => {
      const link = { type } as ExplorerLink;
      if (type === ExplorerType.Sorascan) {
        link.value = `${value}/transaction/${this.value}`;
      } else {
        link.value = `${value}/extrinsic/${this.value}`;
      }
      return link;
    });
  }

  get formattedAddress(): string {
    return formatAddress(this.displayValue, 24);
  }

  handleOpenEtherscan(): void {
    const path = this.type === HashType.EthAccount ? 'address' : 'tx';
    const base = this.soraNetwork !== SoraNetwork.Prod ? 'sepolia' + '.' : '';
    const url = `https://${base}etherscan.io/${path}/${this.value}`;
    const win = window.open(url, '_blank');
    win && win.focus();
  }
}
</script>

<style lang="scss">
.s-dropdown-menu {
  // TODO: fix UI library
  &__item {
    border-radius: calc(var(--s-border-radius-mini) / 2);
  }
}
</style>

<style scoped lang="scss">
$dropdown-right: 15px;
$dropdown-width: var(--s-size-mini);

.s-input-container {
  position: relative;
  + .s-input-container {
    margin-top: var(--s-basic-spacing);
  }
  .s-dropdown-menu {
    position: absolute;
    z-index: 1;
    top: 0;
    right: $dropdown-right;
    bottom: 0;
    margin-top: auto;
    margin-bottom: auto;
    width: $dropdown-width;
    height: var(--s-size-mini);
    line-height: 1;
  }
  .s-button--copy {
    position: absolute;
    top: 0;
    bottom: 0;
    margin-top: auto;
    margin-bottom: auto;
    right: calc(#{$dropdown-right} + #{$dropdown-width} + #{$basic-spacing-mini});
    z-index: 1;
    &,
    &:hover,
    &:focus,
    &:active {
      background-color: transparent;
      border-color: transparent;
    }
  }
}

.transaction-link {
  color: inherit;
  text-decoration: none;
}
</style>
