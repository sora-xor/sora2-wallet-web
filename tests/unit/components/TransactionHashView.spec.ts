import Vuex from 'vuex';
import { shallowMount } from '@vue/test-utils';

import { useDescribe, localVue, i18n } from '../../utils';
import { MOCK_TRANSACTION_HASH_VIEW } from '../../utils/TransactionHashViewMock';

import TransactionHashView from '@/components/TransactionHashView.vue';
import { SoraNetwork } from '@/consts';

const createStore = (env: SoraNetwork) =>
  new Vuex.Store({
    modules: {
      Settings: {
        getters: {
          soraNetwork: () => env,
        },
      },
    } as any,
  });

useDescribe('TransactionHashView.vue', TransactionHashView, () => {
  const formattedAddressLength = 24;
  const ellipsisLength = 3;

  MOCK_TRANSACTION_HASH_VIEW.map((item) =>
    it(`[type: ${item.type}, env: ${SoraNetwork.Dev}]: should be rendered correctly`, () => {
      const wrapper = shallowMount(TransactionHashView, {
        localVue,
        i18n,
        propsData: item,
        store: createStore(SoraNetwork.Dev),
      });
      const txLinks = wrapper
        .findAll('.s-input-container .transaction-link')
        .wrappers.map((item) => item.element) as Array<HTMLAnchorElement>;
      const txInputValue = wrapper.find('.s-input-container s-input-stub').props().value as string;
      expect(wrapper.element).toMatchSnapshot();
      expect(txLinks.length).toBe(1);
      expect(txInputValue.length).toBe(formattedAddressLength + ellipsisLength);
    })
  );

  MOCK_TRANSACTION_HASH_VIEW.map((item) =>
    it(`[type: ${item.type}, env: ${SoraNetwork.Prod}]: should be rendered correctly`, () => {
      const wrapper = shallowMount(TransactionHashView, {
        localVue,
        i18n,
        propsData: item,
        store: createStore(SoraNetwork.Prod),
      });
      const txLinks = wrapper
        .findAll('.s-input-container .transaction-link')
        .wrappers.map((item) => item.element) as Array<HTMLAnchorElement>;
      const txInputValue = wrapper.find('.s-input-container s-input-stub').props().value as string;
      expect(wrapper.element).toMatchSnapshot();
      expect(txLinks.length).toBe(2);
      expect(txInputValue.length).toBe(formattedAddressLength + ellipsisLength);
    })
  );
});
