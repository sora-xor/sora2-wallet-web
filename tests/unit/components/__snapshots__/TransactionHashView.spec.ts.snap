// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionHashView.vue [type: account, env: Dev]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="From"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="cnToWyQbAUWy...yLGKFBTRYbz8"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy From"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <!---->
</div>
`;

exports[`TransactionHashView.vue [type: account, env: Dev]: should be rendered correctly 2`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="To"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="cnToWyQbAUWy...yLGKFBTRYbz8"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy To"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <!---->
</div>
`;

exports[`TransactionHashView.vue [type: account, env: Prod]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="From"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="cnToWyQbAUWy...yLGKFBTRYbz8"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy with-dropdown"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy From"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <s-dropdown-stub
    borderradius="mini"
    buttontype="secondary"
    class="s-dropdown-menu"
    hideonclick="true"
    hidetimeout="150"
    icon="basic-more-vertical-24"
    placement="bottom-end"
    popperclass=""
    showtimeout="250"
    size="medium"
    tabindex="0"
    trigger="hover"
    type="ellipsis"
  >
    <template>
      <a
        class="transaction-link"
        href="https://sora.subscan.io/account/cnToWyQbAUWygF6utv6vSqFfi6VKayKsvXEkbyLGKFBTRYbz8"
        rel="nofollow noopener"
        target="_blank"
      >
        <s-dropdown-item-stub
          class="s-dropdown-menu__item"
        >
          
            View in Subscan
          
        </s-dropdown-item-stub>
      </a>
    </template>
  </s-dropdown-stub>
</div>
`;

exports[`TransactionHashView.vue [type: account, env: Prod]: should be rendered correctly 2`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="To"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="cnToWyQbAUWy...yLGKFBTRYbz8"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy with-dropdown"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy To"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <s-dropdown-stub
    borderradius="mini"
    buttontype="secondary"
    class="s-dropdown-menu"
    hideonclick="true"
    hidetimeout="150"
    icon="basic-more-vertical-24"
    placement="bottom-end"
    popperclass=""
    showtimeout="250"
    size="medium"
    tabindex="0"
    trigger="hover"
    type="ellipsis"
  >
    <template>
      <a
        class="transaction-link"
        href="https://sora.subscan.io/account/cnToWyQbAUWygF6utv6vSqFfi6VKayKsvXEkbyLGKFBTRYbz8"
        rel="nofollow noopener"
        target="_blank"
      >
        <s-dropdown-item-stub
          class="s-dropdown-menu__item"
        >
          
            View in Subscan
          
        </s-dropdown-item-stub>
      </a>
    </template>
  </s-dropdown-stub>
</div>
`;

exports[`TransactionHashView.vue [type: block, env: Dev]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="Block Id"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="0xB8c77482e4...52C74426C6D7"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy Block Id"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <!---->
</div>
`;

exports[`TransactionHashView.vue [type: block, env: Prod]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="Block Id"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="0xB8c77482e4...52C74426C6D7"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy with-dropdown"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy Block Id"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <s-dropdown-stub
    borderradius="mini"
    buttontype="secondary"
    class="s-dropdown-menu"
    hideonclick="true"
    hidetimeout="150"
    icon="basic-more-vertical-24"
    placement="bottom-end"
    popperclass=""
    showtimeout="250"
    size="medium"
    tabindex="0"
    trigger="hover"
    type="ellipsis"
  >
    <template>
      <a
        class="transaction-link"
        href="https://sora.subscan.io/block/0xB8c77482e45F1F4d123DeRwQ5F52C74426C6D7"
        rel="nofollow noopener"
        target="_blank"
      >
        <s-dropdown-item-stub
          class="s-dropdown-menu__item"
        >
          
            View in Subscan
          
        </s-dropdown-item-stub>
      </a>
    </template>
  </s-dropdown-stub>
</div>
`;

exports[`TransactionHashView.vue [type: ethAccount, env: Dev]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="To"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="0x6F984C337a...42cef462e3ff"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy with-dropdown"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy To"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <s-dropdown-stub
    borderradius="mini"
    buttontype="secondary"
    class="s-dropdown-menu"
    hideonclick="true"
    hidetimeout="150"
    icon="basic-more-vertical-24"
    placement="bottom-end"
    popperclass=""
    showtimeout="250"
    size="medium"
    tabindex="0"
    trigger="hover"
    type="ellipsis"
  >
    <template>
      <s-dropdown-item-stub
        class="s-dropdown-menu__item"
      >
        <span>
          View in Etherscan
        </span>
      </s-dropdown-item-stub>
    </template>
  </s-dropdown-stub>
</div>
`;

exports[`TransactionHashView.vue [type: ethAccount, env: Prod]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="To"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="0x6F984C337a...42cef462e3ff"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy with-dropdown"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy To"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <s-dropdown-stub
    borderradius="mini"
    buttontype="secondary"
    class="s-dropdown-menu"
    hideonclick="true"
    hidetimeout="150"
    icon="basic-more-vertical-24"
    placement="bottom-end"
    popperclass=""
    showtimeout="250"
    size="medium"
    tabindex="0"
    trigger="hover"
    type="ellipsis"
  >
    <template>
      <s-dropdown-item-stub
        class="s-dropdown-menu__item"
      >
        <span>
          View in Etherscan
        </span>
      </s-dropdown-item-stub>
    </template>
  </s-dropdown-stub>
</div>
`;

exports[`TransactionHashView.vue [type: ethTransaction, env: Dev]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="Transaction Id"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="0x07f7ec4c5f...87b8674b2e80"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy with-dropdown"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy Transaction Id"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <s-dropdown-stub
    borderradius="mini"
    buttontype="secondary"
    class="s-dropdown-menu"
    hideonclick="true"
    hidetimeout="150"
    icon="basic-more-vertical-24"
    placement="bottom-end"
    popperclass=""
    showtimeout="250"
    size="medium"
    tabindex="0"
    trigger="hover"
    type="ellipsis"
  >
    <template>
      <s-dropdown-item-stub
        class="s-dropdown-menu__item"
      >
        <span>
          View in Etherscan
        </span>
      </s-dropdown-item-stub>
    </template>
  </s-dropdown-stub>
</div>
`;

exports[`TransactionHashView.vue [type: ethTransaction, env: Prod]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="Transaction Id"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="0x07f7ec4c5f...87b8674b2e80"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy with-dropdown"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy Transaction Id"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <s-dropdown-stub
    borderradius="mini"
    buttontype="secondary"
    class="s-dropdown-menu"
    hideonclick="true"
    hidetimeout="150"
    icon="basic-more-vertical-24"
    placement="bottom-end"
    popperclass=""
    showtimeout="250"
    size="medium"
    tabindex="0"
    trigger="hover"
    type="ellipsis"
  >
    <template>
      <s-dropdown-item-stub
        class="s-dropdown-menu__item"
      >
        <span>
          View in Etherscan
        </span>
      </s-dropdown-item-stub>
    </template>
  </s-dropdown-stub>
</div>
`;

exports[`TransactionHashView.vue [type: id, env: Dev]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="Transaction Id"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="0xB8c77482e4...52C74426C6D7"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy Transaction Id"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <!---->
</div>
`;

exports[`TransactionHashView.vue [type: id, env: Prod]: should be rendered correctly 1`] = `
<div
  class="s-input-container"
>
  <s-input-stub
    accept=""
    autocomplete="off"
    borderradius="small"
    form=""
    label=""
    name=""
    placeholder="Transaction Id"
    prefix=""
    readonly="true"
    rows="2"
    size="big"
    suffix=""
    tabindex="-1"
    type="text"
    usedesignsystem="true"
    value="0xB8c77482e4...52C74426C6D7"
  />
   
  <s-button-stub
    alternative="true"
    borderradius="small"
    class="s-button--copy with-dropdown"
    icon="basic-copy-24"
    iconposition="left"
    nativetype="button"
    size="medium"
    tabindex="0"
    tooltip="Copy Transaction Id"
    tooltipplacement="top"
    type="action"
    usedesignsystem="true"
  />
   
  <s-dropdown-stub
    borderradius="mini"
    buttontype="secondary"
    class="s-dropdown-menu"
    hideonclick="true"
    hidetimeout="150"
    icon="basic-more-vertical-24"
    placement="bottom-end"
    popperclass=""
    showtimeout="250"
    size="medium"
    tabindex="0"
    trigger="hover"
    type="ellipsis"
  >
    <template>
      <a
        class="transaction-link"
        href="https://sora.subscan.io/extrinsic/0xB8c77482e45F1F4d123DeRwQ5F52C74426C6D7"
        rel="nofollow noopener"
        target="_blank"
      >
        <s-dropdown-item-stub
          class="s-dropdown-menu__item"
        >
          
            View in Subscan
          
        </s-dropdown-item-stub>
      </a>
    </template>
  </s-dropdown-stub>
</div>
`;
